mono/metadata/boehm-gc.c

```diff
@@ -668,6 +668,8 @@ mono_gc_alloc_obj (MonoVTable *vtable, size_t size)
 	if (G_UNLIKELY (mono_profiler_events & MONO_PROFILE_ALLOCATIONS))
 		mono_profiler_allocation (obj);
 
+    mono_stats.new_object_count++;
```

Comment:
Should these be atomic? And do they need to be behind any #if checks related to stats being enabled?